ern-local-cli/src/commands/create-api.js

```diff
@@ -38,54 +43,63 @@ exports.builder = function (yargs: any) {
 exports.handler = async function ({
   apiName,
   scope,
+  packageName,
   apiVersion,
   apiAuthor,
   schemaPath,
   skipNpmCheck
 } : {
   apiName: string,
   scope?: string,
+  packageName: string,
   apiVersion?: string,
   apiAuthor?: string,
   schemaPath?: string,
   skipNpmCheck? : boolean
 }) {
-  // Check if the api name is valid npm package name
-  // https://docs.npmjs.com/files/package.json
-  await utils.logErrorAndExitIfNotSatisfied({
-    isValidNpmPackageName: {
-      name: apiName
+  try {
+    await utils.logErrorAndExitIfNotSatisfied({
+      isValidElectrodeNativeModuleName: {
+        name: apiName
+      }
+    })
+
+    // Construct the package name
+    if (!packageName) {
+      packageName = core.getDefaultPackageNameForModule(apiName, ModuleTypes.API)
```

Comment:
Could you reuse the same logic as in `create-miniapp`
i.e, if the `packageName` is not explicitly provided as a command argument, the default generated name is just the default suggestion, we still prompt the user if he/she wants to use this name or a different one

https://github.com/electrode-io/electrode-native/blob/master/ern-local-cli/src/commands/create-miniapp.js#L62

You can also copy/paste the `promptForPackageName` function (just replace `MiniApp` by `API` in the message string). 

The same way will be done for create-api-impl so that the create commands are properly aligned and consistent.

As they will share most of the same logic it makes it easier to refactor (regroup logic) later on as well.

Reply:
@belemaire by design for create-api and create-api-impl I thought it's better to avoid prompt  for default package name. [documentation states the default name convention]
this way we avoid adding additional option to avoid CI prompting etc. It's generated code IMHO and we also have introduce a flag if user wishes to override default name for package. Thoughts please?

Reply:
nvm . ignore the comment. I think we want to align the logic across so makes sense.